### DISCUSSION TITLE
v3.1.0 (Beta)

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#3624] [Releases]
## Installation
Visit [https://getutm.app/install/][1] for the most up to date installation instructions.

## Highlights
* Wizard redesign (thanks @js-john)
* Store .utm bundles anywhere on disk (including on external drives) and import them as a shortcut. See notes below for some limitations.

## Notes
* (iOS) **UTM v3.1.x will be the last update to support iOS 11 and iOS 12.** If you are still running those versions, please make sure you report any issues now so they can be fixed by the final release of v3.1.x. After that, iOS 13+ will be required to run future versions of UTM.
* This update makes significant changes to the backend of UTM which may have introduced more issues than usual. As a precaution, please make sure you back up your VMs before running this beta in case there are bugs that cause data loss.
* You can now run .utm files stored anywhere. There are a few caveats: First, all newly created VMs will still be created in the default storage (app sandbox). You can move the VM after creation with the new button on the toolbar or by right clicking (or force touch) on the VM entry and selecting "Move..." Second, when VM shortcuts are deleted, the underlying data is unaffected, and shortcuts are automatically deleted when they are no longer valid (for example, the .utm was moved). You can re-import the .utm by either double clicking it from Finder, or using "File -> Import Virtual Machine..." Finally, Apple VMs do not support persistent shortcuts, which means the shortcut is *always* deleted after UTM quits. This means you must re-import Apple VMs every time UTM is launched.

## Changes (v3.1.0)
* Improved design of wizard on iOS (thanks @js-john)
* Updated zh-Hans translation (thanks @ty-yqs)
* Refactored backend renderer into [CocoaSpice](https://github.com/utmapp/CocoaSpice)
* Updated backend code to use Swift 5.5 concurrency features
* In Apple VM configs, do not discard non-persistent settings (shared directories) when canceling out of settings window
* Add option to run VM as a shortcut (see notes above) (#132, #1205)
* Added icon for Pop! OS to picker (thanks @conath)
* Fixed QEMU custom argument parsing when there is quoted space in middle of the argument (#3579)
* Update the VM screenshot on a timer (#3499)
* Added new global setting to disable saving screenshot to disk
* Improved synchronization of state from the VM window and the main window (play button visible when stopped, busy spinner visible when starting, etc)
* (iOS) Fix crash while installing on older versions of jailbroken iOS (#3543)
* (macOS) Fix an issue where Apple VM aux storage was not created when switching from Linux to macOS boot option
* (macOS) Ability to move a .utm VM to external storage
* (macOS) Toolbar option to delete VM should be disabled when VM is running
* (macOS) Support taking screenshots of Apple VMs

## Issues
Please check the full list on [Github][2] and help report any bug you find that is not listed.

[1]: https://getutm.app/install/
[2]: https://github.com/utmapp/UTM/issues

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v3.1.0'>v3.1.0 (Beta)</a>.</em>

**Comment by kkebo** [comment]
> You can now run .utm files stored anywhere.

Is this a feature only for macOS?

**Reply by osy** [reply]
Not strictly no but I haven't tested it

**Reply by kkebo** [reply]
Thanks. It seemed to work when I opened a .utm file from the share sheet on iOS. However, I think that the instructions in the release notes are only for macOS.

> You can now run .utm files stored anywhere. There are a few caveats: First, all newly created VMs will still be created in the default storage (app sandbox). You can move the VM after creation with the new button on the toolbar or by right clicking (or force touch) on the VM entry and selecting "Move..." Second, when VM shortcuts are deleted, the underlying data is unaffected, and shortcuts are automatically deleted when they are no longer valid (for example, the .utm was moved). You can re-import the .utm by either double clicking it from Finder, or using "File -> Import Virtual Machine..." Finally, Apple VMs do not support persistent shortcuts, which means the shortcut is always deleted after UTM quits. This means you must re-import Apple VMs every time UTM is launched.

**Reply by osy** [reply]
I didn’t mention iOS because I didn’t test it. It’ll be in the next release once I do.

**Comment by brunocastello** [comment]
This is an amazing release. I am now very happy to be able to run VMs from my external storage without workarounds. Thanks, @osy ! You never cease to amaze me with every release.

**Comment by brunocastello** [comment]
Did anyone notice that the operating systems icons are a bit squeezed/distorted now? I was thinking about creating custom ones for myself to set up there but I don't know the exact size they should have.

**Reply by conath** [reply]
128x128px

**Comment by theanuraaag** [comment]
I am getting black screen when I tried to install Kali Linux on my Macbook Air M1 using UTM , it works normal till the option install, graphical install, etc but when I click on Graphical Install a black screen appears and  nothing happen after that .
<img width="1440" alt="Screenshot 2022-05-18 at 11 31 28 AM" src="https://user-images.githubusercontent.com/86941924/168968310-7d79bd34-59c1-4f56-9c0a-6e8cf80633b8.png">
<img width="1440" alt="Screenshot 2022-05-18 at 11 31 36 AM" src="https://user-images.githubusercontent.com/86941924/168968320-da2b90da-dfc2-4c38-bf39-a6fdfca4c5cb.png">
<img width="1440" alt="Screenshot 2022-05-18 at 11 32 51 AM" src="https://user-images.githubusercontent.com/86941924/168968345-11758afd-751f-4507-a388-f08e7b096255.png">

**Reply by conath** [reply]
You could try again with my "Ubuntu 21.10 (ARM aarch64)" template from my website: https://chrisp.cafe/UTM/

It should work for Kali Linux ARM as well.

**Reply by theanuraaag** [reply]
Can you please make a pre installed file of Kali Linux ?

**Reply by conath** [reply]
@theanuraaag It looks like Kali is not offering an ARM64 version you can use in UTM, though I may be wrong and just confused by their website. Anyways there is a live boot option available that I would recommend for you, it means you flash the image onto a USB stick and boot your Mac from that: https://www.kali.org/get-kali/#kali-live

**Reply by theanuraaag** [reply]
@conath but I think it will be just temporary ?
Whenever I remove USB it will get reset ?

**Reply by conath** [reply]
I'm not sure- there is probably more information on the Kali Linux website.

**Reply by theanuraaag** [reply]
Ok thanks for your time @conath  !
